src/net_processing.cpp

```diff
@@ -1230,7 +1230,7 @@ void PeerManagerImpl::InitializeNode(CNode *pnode)
         mapNodeState.emplace_hint(mapNodeState.end(), std::piecewise_construct, std::forward_as_tuple(nodeid), std::forward_as_tuple(pnode->IsInboundConn()));
         assert(m_txrequest.Count(nodeid) == 0);
     }
-    PeerRef peer = std::make_shared<Peer>(nodeid, /*tx_relay=*/ !pnode->IsBlockOnlyConn());
+    PeerRef peer = std::make_shared<Peer>(nodeid, /*tx_relay=*/!pnode->IsBlockOnlyConn() && !pnode->IsManualBlockRelayConn());
```

Comment:
Why not directly change IsBlockOnlyConn() directly as below, then can avoid to define new function IsManualBlockRelayConn()?

     bool IsBlockOnlyConn() const {
-        return m_conn_type == ConnectionType::BLOCK_RELAY;
+        return m_conn_type == ConnectionType::BLOCK_RELAY || m_conn_type == ConnectionType::MANUAL_BLOCK_RELAY;
     }

Reply:
See https://github.com/bitcoin/bitcoin/pull/24170#discussion_r795710002 - there are several placed that use `IsBlockOnlyConn()` with logic specific to automatic block-relay-only connections that have logic that does not apply to manual ones. So if I did that, we'd need additional logic in these places to distinguish between manual and automatic connections.

Reply:
That seems preferable?

Reply:
I have changed the logic now such that `IsBlockRelayConn` is used when both connection types are meant, where as `IsManualBlockRelayConn` and `IsAutomaticBlockRelayConn` are used in spots where only one is meant.